### ISSUE TITLE
[feature] Create the artifact file by builder

### ISSUE THREAD

**Issue by zinmoon** [slsa-framework/slsa-github-generator#1127]
The slsa-github-generator has three builders.
Container, generic and go builder

In the case of go builder, the artifact/provenance file are created by the tool.
On the other hand, other builders only create a provenance.

In my opinion, the artifact file should be created as given build command and environment by the tool.
Through this operation, we can trust the build command and the environment in provenance.

Why other builders don't support the creating artifacts?
Do you have any plan to support it for other language such as python

**Comment by ianlewis** [comment]
> In the case of go builder, the artifact/provenance file are created by the tool. On the other hand, other builders only create a provenance.
> 
> In my opinion, the artifact file should be created as given build command and environment by the tool. Through this operation, we can trust the build command and the environment in provenance.

Yes, you do get a bit better guarantees if the reusable workflow does the build. The generator workflows are given the subject by the user workflow. However, the generator workflows are able to accurately link to the user's workflow at a specific git digest. This can allow consumers of the provenance to know what specific workflow was used to build the artifacts and inspect it.

> Why other builders don't support the creating artifacts? Do you have any plan to support it for other language such as python

We have plans to provide tools like actions and reusable workflows to allow different ecosystems to create their own builders. We plan to help them do so. This would include python. This is tracked by #763 but development hasn't started yet.
